code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm

```diff
@@ -2,1474 +2,1217 @@
+					xeno_stats.landslide_damage += vehicle_damage
+					atoms_hit += V
 					continue
-				hit_object(affected_object)
-	steps_to_take--
-	step(xeno_owner, direction, 1)
-	if(steps_to_take <= 2)
-		speed += 0.5
-		new /obj/effect/temp_visual/behemoth/landslide/dust/charge(owner_turf, direction)
-	addtimer(CALLBACK(src, PROC_REF(enhanced_do_charge), direction, damage, speed, steps_to_take), speed)
-
-/**
- * Ends the charge.
- * * reason: If specified, determines the reason why the charge ended, and does the respective balloon alert. Leave empty for no reason.
-*/
-/datum/action/ability/activable/xeno/landslide/proc/end_charge(reason)
-	ability_active = FALSE
-	REMOVE_TRAIT(owner, TRAIT_SILENT_FOOTSTEPS, XENO_TRAIT)
-	xeno_owner.behemoth_charging = FALSE
-	if(!xeno_owner.lying_angle)
-		xeno_owner.set_canmove(TRUE)
-	add_cooldown()
-	switch(reason)
-		if(LANDSLIDE_ENDED_CANCELLED) // The user manually cancelled the ability at some point during its use.
-			xeno_owner.balloon_alert(xeno_owner, "cancelled!")
-		if(LANDSLIDE_ENDED_NO_PLASMA) // During the charge, the user did not have enough plasma to maintain the ability.
-			xeno_owner.balloon_alert(xeno_owner, "insufficient plasma!")
-
-/**
- * Applies several effects to a living target.
- * * living_target: The targeted living mob.
- * * damage: The damage inflicted by related effects.
-*/
-/datum/action/ability/activable/xeno/landslide/proc/hit_living(mob/living/living_target, damage)
-	if(!living_target || !damage)
-		return
-	if(living_target.buckled)
-		living_target.buckled.unbuckle_mob(living_target, TRUE)
-	if(!living_target.lying_angle)
-		living_target.Knockdown(LANDSLIDE_KNOCKDOWN_DURATION)
-		new /obj/effect/temp_visual/behemoth/landslide/hit(get_turf(living_target))
-		playsound(living_target, 'sound/effects/alien/behemoth/landslide_hit_mob.ogg', 30, TRUE)
-	living_target.emote("scream")
-	shake_camera(living_target, 1, 0.8)
-	living_target.apply_damage(damage, BRUTE, blocked = MELEE, attacker = owner)
-
-/**
- * Attempts to deconstruct the object in question if possible.
- * * object_target: The targeted object.
-*/
-/datum/action/ability/activable/xeno/landslide/proc/hit_object(obj/object_target)
-	if(!object_target)
-		return
-	var/object_turf = get_turf(object_target)
-	if(istype(object_target, /obj/machinery/vending))
-		var/obj/machinery/vending/vending_target = object_target
-		playsound(object_turf, 'sound/effects/meteorimpact.ogg', 30, TRUE)
-		new /obj/effect/temp_visual/behemoth/landslide/hit(object_turf)
-		vending_target.tip_over()
-		return
-	if(istype(object_target, /obj/structure/reagent_dispensers/fueltank))
-		var/obj/structure/reagent_dispensers/fueltank/tank_target = object_target
-		tank_target.explode()
+
+/// Adds effects for when we make impact against something.
+/datum/action/ability/activable/xeno/landslide/proc/on_impact(datum/source, atom/A, speed)
+	SIGNAL_HANDLER
+	var/turf/T = get_turf(A)
```

Comment:
going to stop commenting on every instance one 1 letter vars but please fix it

Reply:
Changed all one letter vars to be more detailed. If I missed any, let me know.